ocaml/typing/typecore.ml

```diff
@@ -5045,6 +5045,41 @@ let add_check_attribute expr attributes =
     end
   | _ -> expr
 
+let zero_alloc_of_application ~num_args attrs funct =
+  let zero_alloc =
+    Builtin_attributes.get_property_attribute ~in_signature:false
+      ~default_arity:num_args attrs Zero_alloc
+  in
+  let zero_alloc =
+    match zero_alloc with
+    | Assume _ | Ignore_assert_all _ | Check _ ->
+      (* The user wrote a zero_alloc attribute on the application - keep it.
+         (Note that `ignore` and `check` aren't really allowed here, and will be
+         rejected by the call to `Builtin_attributes.assume_zero_alloc` below.)
+       *)
+      zero_alloc
+    | Default_check ->
+      (* No zero_alloc attribute on the application - see if we can find
+         information from the value description. *)
```

Comment:
I find this comment to be less useful than an explanation  of what's going on with `use_opt` and `c.opt`. I might comment that instead.

Something like:

"We assume the call is zero-alloc if the function is known to be zero-alloc. If the function is zero-alloc opt, then we need to be sure that the opt checks were run to license this assumption. We judge whether the opt checks were run using the presence of the `zero_alloc_check` command line flag."

Reply:
Yes, another slightly out-of-date comment.  I've updated it, thanks.